localization/ekf_localizer/test/test_covariance.cpp

```diff
@@ -0,0 +1,79 @@
+// Copyright 2022 Autoware Foundation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "ekf_localizer/covariance.hpp"
+
+#include <gtest/gtest.h>
+
+TEST(EKFCovarianceToPoseMessageCovariance, SmokeTest)
+{
+  {
+    Matrix6d P = Matrix6d::Zero();
+    P(0, 0) = 1.;
+    P(0, 1) = 2.;
+    P(0, 2) = 3.;
+    P(1, 0) = 4.;
+    P(1, 1) = 5.;
+    P(1, 2) = 6.;
+    P(2, 0) = 7.;
+    P(2, 1) = 8.;
+    P(2, 2) = 9.;
+
+    std::array<double, 36> covariance = ekfCovarianceToPoseMessageCovariance(P);
+    EXPECT_EQ(covariance(0), 1.);
+    EXPECT_EQ(covariance(1), 2.);
+    EXPECT_EQ(covariance(2), 3.);
```

Comment:
Does this index need to be 5?
```suggestion
    EXPECT_EQ(covariance(5), 3.);
```

Reply:
How about using `COV_IDX::` or `IDX::` here as well?

Or, I think we should at least leave some comments to notify the readers that these indices are dependent on the definition of EKF state (like `EXPECT_EQ(covariance(2), 3.); // testing COV_IDX::X_YAW` for each line).

Reply:
Still I'm not sure why we need to use the COV_IDX struct. We can just refer to the ROS covariance definition.

Reply:
Actually I agree that `COV_IDX` may not be necessary, but I also think that the indices "0, 1, 5, 6, 7, 11, ..." is also a bit confusing. (I would rather prefer the following style.)
```
...[0 * 6 + 0]
...[0 * 6 + 1]
...[0 * 6 + 5]
...[1 * 6 + 0]
...[1 * 6 + 1]
...
```

I will leave the final decision up to you.

Reply:
The test should be independent of the IDX or COV_IDX covariance so I leave it there